### PR TITLE
Start accommodating error and received emails for form submissions

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR adds two new types of notifications from Simple Forms API alongside the existing `confirmation` email notifications: `error` and `received`. It also renames the `ConfirmationEmail` to a more generic name, `NotificationEmail`. I'd like to merge this as-is, only affecting one form (21-10210), because if I roll out the changes across all our forms at once, it'll be a huge diff. I can follow up with more PRs to convert the rest of our forms after this one.

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1668

## Testing done
Tests are extended and updated.

### CODE REVIEW

File: modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
+      'vba_21_10210' => {
+        confirmation: Settings.vanotify.services.va_gov.template_id.form21_10210_confirmation_email,
+        error: Settings.vanotify.services.va_gov.template_id.form21_10210_error_email,
+        received: Settings.vanotify.services.va_gov.template_id.form21_10210_received_email
+      },
+      'vba_20_10206' => { confirmation: Settings.vanotify.services.va_gov.template_id.form20_10206_confirmation_email },
+      'vba_20_10207' => { confirmation: Settings.vanotify.services.va_gov.template_id.form20_10207_confirmation_email },
+      'vba_40_0247' => { confirmation: Settings.vanotify.services.va_gov.template_id.form40_0247_confirmation_email }
     }.freeze
     SUPPORTED_FORMS = TEMPLATE_IDS.keys
```
Comment (pennja) [inline modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:22]: My mind can't help but go to dynamically generating this list. I'd love your thoughts on something like this?

```ruby
TEMPLATE_IDS = build_template_ids.freeze
SUPPORTED_FORMS = TEMPLATE_IDS.keys

...

private

def build_template_ids
  form_templates = {
    'vba_21_0845' => %i[confirmation],
    'vba_21p_0847' => %i[confirmation],
    'vba_21_0966' => %i[confirmation],
    'vba_21_0972' => %i[confirmation],
    'vba_21_4142' => %i[confirmation],
    'vba_21_10210' => %i[confirmation error received],
    'vba_20_10206' => %i[confirmation],
    'vba_20_10207' => %i[confirmation],
    'vba_40_0247' => %i[confirmation]
  }

  form_templates.each_with_object({}) do |(form, template_types), hash|
    hash[form] = template_types.each_with_object({}) do |type, inner_hash|
      inner_hash[type] = Settings.vanotify.services.va_gov.template_id.public_send("#{form}_#{type}_email")
    end
  end
end
```

Comment (pennja) [reply modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:22]: Actually, I've decided I'm getting too "fancy" with it. Disregard 😂

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍

File: modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb (hunk 1 above)
Comment (Thrillberg) [reply modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb:22]: 👀 ooo that's interesting! I think it's a little much for this configuration hash. I think there's benefit to being able to see it all clearly (even if verbosely). But its a nice idea!

**Review by ryan-mcneil — APPROVED** [review]

**Comment by ryan-mcneil** [comment]
@Thrillberg I'm going to have to merge it because of the weird CI error. please ping me if/when you want me to do so

**Comment by bdech** [comment]
@ryan-mcneil, Eric said we're good to merge

**Pull request merged**